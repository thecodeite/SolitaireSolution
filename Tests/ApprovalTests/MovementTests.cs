using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using Solitaire.Models;

namespace Tests.ApprovalTests
{

    [TestFixture]
    [UseReporter(typeof (DiffReporter))]
    //[UseReporter(typeof(VisualStudioReporter))]
    internal class MovementTests
    {
        [Test]
        public void can_make_legal_move_from_one_column_to_another()
        {
            var game = new Game();
            game.Deal(Deck.MakeTestDeck());

            // Move jack of clubs from column 6 to column 7
            game.ParseInput("cJ 7");

            string board = game.Render();
            Approvals.Verify(board);
        }
        
        [Test]
        public void can_make_legal_move_from_one_column_to_another_and_child_cards_move_too()
        {
            var game = new Game();
            game.Deal(Deck.MakeTestDeck());

            // Move jack of clubs from column 6 to column 7
            game.ParseInput("cJ 7");

            // Move queene of diamonds from column 7 to column 2
            game.ParseInput("DQ 2");

            string board = game.Render();
            Approvals.Verify(board);
        }

        [Test]
        public void can_make_legal_move_ace_from_column_to_waste()
        {
            var game = new Game();
            game.Deal(Deck.MakeTestDeck());

            // Move ace of hearts to hearts discard pile
            game.ParseInput("HA H");

            string board = game.Render();
            Approvals.Verify(board);
        }

        [Test]
        public void can_make_legal_move_king_to_empty_space()
        {
            var game = new Game();
            game.Deal(Deck.MakeTestDeck());

            // Move ace of hearts to hearts discard pile
            game.ParseInput("HA H");

            // Move king of X to column 1
            game.ParseInput("sK 1");

            string board = game.Render();
            Approvals.Verify(board);
        }

        [Test]
        public void can_make_legal_move_card_from_stack_to_columns()
        {
            var game = new Game();
            game.Deal(Deck.MakeTestDeck());

            // Move two of diamonds from stack to column 4
            game.ParseInput("D2 4");

            string board = game.Render();
            Approvals.Verify(board);
        }
    }
}
