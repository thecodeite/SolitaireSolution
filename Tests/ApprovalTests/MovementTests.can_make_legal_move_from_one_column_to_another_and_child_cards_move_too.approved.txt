 ColumnNames   S[T]ack        [1] [2] [3] [4] [5] [6] [7] [D] [H] [C] [S]
-------------------------------------------------------------------------
                   D2         HA  **  **  **  **  **  **                 
                                  sK  **  **  **  **  **                 
                                  DQ  D8  **  **  **  **                 
                                  cJ      c3  **  **  **                 
                                              D6  HJ  **                 
                                                      cA                 
